lib/org/chromium/webidl/ParserRunner.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'ParserRunner',
+  extends: 'foam.box.Runnable',
+
+  requires: ['org.chromium.webidl.PipelineMessage'],
+
+  documentation: 'Runnable Box that parsers IDLFileContents into ASTs.',
+
+  methods: [
+    function run(message) {
+      if (!this.PipelineMessage.isInstance(message))
+        return this.error(new Error("ParserRunner: Run() expects a" +
+            " PipelineMessage object!"));
+      else if (!message.idlFile || !message.parser)
+        return this.error(new Error("ParserRunner: Missing required" +
+            " properties in PipelineMessage!"));
+
+      var file = message.idlFile;
+      var parser = message.parser;
+
+      // Load appropriate parser depending on configuration.
+      var parser = foam.lookup(`org.chromium.webidl.${parser}`).create();
+      var results = parser.parseString(file.contents);
+
+      // Determine if parse of entire file was successful.
+      if (file.contents.length !== results.pos) {
```

Comment:
A failed parse returns either `undefined` or `null`. (Plz double-check `foam2/src/foam/parse/parse.js`). We should check `.value` for a "parse failed" value here too. We don't want to consume the whole input, but fail to parse, and assume that parsing was successful.